### PR TITLE
Don't loose issues on analysis errors

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/tests/common/WorkspaceHelpers.java [hunk 1]
```diff
@@ -26,28 +26,57 @@
...
+        doCleanWorkspace();
+      } catch (InterruptedException e) {
+        throw e;
+      } catch (OperationCanceledException e) {
+        throw e;
+      } catch (Exception e) {
+        cause = e;
+        e.printStackTrace();
+        System.out.println(i);
+        Thread.sleep(6 * 1000);
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/tests/common/WorkspaceHelpers.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2925)

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java [hunk 2]
```diff
@@ -453,88 +465,21 @@ public void uncaughtException(Thread th, Throwable ex) {
...
     if (startLine == null) {
       checksum = null;
     } else {
-      Range rangeInFile = SonarMarker.findRangeInFile(issue, iDoc);
-      checksum = SonarMarker.checksum(rangeInFile.getContent());
+      Range rangeInFile;
+      try {
+        rangeInFile = SonarMarker.findRangeInFile(issue, iDoc);
+        checksum = SonarMarker.checksum(rangeInFile.getContent());
+      } catch (BadLocationException e) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* MAJOR 4 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AnalyzeCommand.java#L111](https://github.com/SonarSource/sonarlint-eclipse/blob/19bbbfb1ca83878d3dc6d0dc836414ce0820efbe/org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/command/AnalyzeCommand.java#L111): Make this anonymous inner class a lambda [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1604)
1. MAJOR [AnalyzeCommand.java#L151](https://github.com/SonarSource/sonarlint-eclipse/blob/19bbbfb1ca83878d3dc6d0dc836414ce0820efbe/org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/command/AnalyzeCommand.java#L151): Make this anonymous inner class a lambda [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1604)
1. MINOR [TrackableIssue.java](https://github.com/SonarSource/sonarlint-eclipse/blob/19bbbfb1ca83878d3dc6d0dc836414ce0820efbe/org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/TrackableIssue.java): Add a new line at the end of this file. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00113)
1. MINOR [TrackableMarker.java](https://github.com/SonarSource/sonarlint-eclipse/blob/19bbbfb1ca83878d3dc6d0dc836414ce0820efbe/org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/TrackableMarker.java): Add a new line at the end of this file. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00113)

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**